flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/diff/DiffResult.java

```diff
@@ -59,18 +63,48 @@ public String toString() {
             return "";
         }
 
-        final ToStringBuilder lhsBuilder =
-                new ToStringBuilder(left, ToStringStyle.SHORT_PREFIX_STYLE);
-        final ToStringBuilder rhsBuilder =
-                new ToStringBuilder(right, ToStringStyle.SHORT_PREFIX_STYLE);
+        final StringBuilder builder = new StringBuilder();
+        builder.append(left.getClass().getSimpleName()).append("[");
 
         diffList.forEach(
                 diff -> {
-                    lhsBuilder.append(diff.getFieldName(), diff.getLeft());
-                    rhsBuilder.append(diff.getFieldName(), diff.getRight());
+                    try {
+                        JsonNode before =
+                                objectMapper.readTree(
+                                        objectMapper.writeValueAsString(diff.getLeft()));
+                        JsonNode after =
+                                objectMapper.readTree(
+                                        objectMapper.writeValueAsString(diff.getRight()));
+                        JsonNode jsonDiff = JsonDiff.asJson(before, after);
+                        jsonDiff.forEach(
+                                row -> {
```

Comment:
that clause should be extracted to another method

Reply:
Again, I am not sure what part do you think would merit its own method?
The block should be quite easy to understand, just an if statement with a bunch of append statements.
I am not sure that putting it into a method would help with readability

Reply:
IMHO,  Extracting multiple lines within a loop to a well named method help others to understand the intention behind that piece of code. it makes the code smaller, help with debugging and error handling when needed.

i.e , obj.forEach(x -> createDiffObject(x))

I see that in that case there is another loop for every row, it would be nicer to have another small method which indicates the operation per row.(IMO)